src/tic_tac_toe/referee.clj

```diff
@@ -11,3 +11,36 @@
 (defn validate-move
   [player move]
   (convert-to-spot move))
+
+(defn draw?
+  [board]
+  (board/full? board))
+
+(defn matching-pattern?
+  [token pattern]
+  (loop [token token
+         pattern pattern]
+    (match pattern
+      [token] true
+      [" " & r] false
+      [(_ :guard #(not= token %)) & r] false
+      [token & r] (recur token (vec (rest pattern))))))
+
+(defn winner?
+  [board]
+  (let [patterns (board/give-patterns board)]
+    (loop [patterns patterns]
+      (cond
+        (empty? patterns) false
+        (matching-pattern? ((first patterns) 0) (vec (first patterns))) true
```

Comment:
I am not sure what is going on here? Could make it more clear?

Reply:
I broke it down.  `matching-pattern?` is still in place, but uses [smaller functions](https://github.com/saralein/tic-tac-toe_clojure/pull/6/files#diff-f014d0e7db79dee6fc42af72f74d5325R54) which I think should be easier to read.  Changed `get-status` to `get-winner`, which now shares some underlying functionality with `winner?`.